nvflare/apis/impl/controller.py

```diff
@@ -849,7 +850,9 @@ def _check_tasks(self):
                     continue
 
                 # check whether clients that the task is waiting are all dead
-                if not self._task_has_running_client(task):
+                dead_client = self._get_task_dead_client(task)
```

Comment:
I thought about that too - adding a new status. We could also think this is one case of timeout since the client will never send response. 

Since we will redesign this later (quick abort, ...), I decided to take a simple approach for now.